plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HiveTableOperations.java

```diff
@@ -103,6 +111,15 @@
         this.tableName = requireNonNull(table, "table is null");
         this.owner = requireNonNull(owner, "owner is null");
         this.location = requireNonNull(location, "location is null");
+
+        requireNonNull(icebergTableMetadata, "icebergTableMetadata is null");
+        if (icebergTableMetadata.isPresent()) {
```

Comment:
Since this actually happens after construction and is somewhat independent, we could have another method `initializeFromMetadata(TableMetadata)` instead of an optional constructor argument.